oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -107,20 +107,35 @@ impl Runtime {
     /// Thread safe method for signaling termination to a [`Runtime`] and waiting for its node
     /// threads to terminate.
     pub fn stop(&self) {
-        let mut threads = {
+        // Unpark any threads that are blocked waiting on any channels.
+        for channel in self
+            .channels
+            .channels
+            .read()
+            .expect("could not acquire channel mapping")
+            .values()
+        {
+            for thread in channel
+                .waiting_threads
+                .lock()
+                .expect("could not acquire waiting threads for channel")
+                .values()
+            {
+                if let Some(thread) = thread.upgrade() {
+                    thread.unpark();
+                }
+            }
+        }
+
+        let mut nodes = {
             let mut nodes = self.nodes.lock().unwrap();
             self.terminating.store(true, SeqCst);
```

Comment:
I think swapping the nodes out and setting terminating to true needs to occur before unparking the threads as the wait_on_channels loop will cause them to park again if runtime is not terminating.

Reply:
This is probably why CI is timing out.

Reply:
Good point, fixed, and added some comments, PTAL.